### PR TITLE
Create page about NAT

### PR DESCRIPTION
**Pull request by MichaelDeciso**
This page links to the nptv6 page I wrote earlier (but hasn't been merged yet).

### CODE REVIEW

File: source/manual/nat.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+ Source
+ Destination / invert   Invert match in “Destination” field.
+ Destination
+ Description            A description to easily find the rule in the overview.
+ NAT reflection         See “Some terms explained”. Leave this on the default unless you have a good reason not to.
+====================== =================================================================================================
+
+--------
+Outbound
+--------
```
Comment (fabianfrz) [inline source/manual/nat.rst]: I would add the terms SNAT / Source NAT here as it is commonly called this way (also in netfilter on Linux).

File: source/manual/nat.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+instead of the internal one (which would the most logical), NAT reflection can rewrite this request so that it uses
+the internal IP, in order to avoid taking a detour and applying rules meant for actual outside traffic.
+
+**Pool options**: When there are multiple IPs to choose from, this option will allow regulating which IP gets used.
+The default, Round Robin, will simply distribute packets to one server after the other. If you only have one external
+IP, this option has no effect.
+
+---------------
+Port forwarding
+---------------
```
Comment (fabianfrz) [inline source/manual/nat.rst]: I would add the terms DNAT / Destination NAT here as it is commonly called this way (also in netfilter on Linux).

Should also mention transparent proxies

File: source/manual/nat.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
+===========================
+Network Address Translation
+===========================
+
+Network Address Translation (abbreviated to NAT) is a way to separate external and internal networks (WANs and LANs),
+and to share an external IP between clients on the interal network. NAT only works on IPv4. For IPv6,
+:doc:`Network Prefix Translation <nptv6>` can be used instead.
```
Comment (fabianfrz) [inline source/manual/nat.rst]: NAT64 does exist as well in IPv6

File: source/manual/nat.rst [hunk 4]
```diff
@@ -0,0 +1,139 @@
+===========================
+Network Address Translation
+===========================
+
```
Comment (fabianfrz) [inline source/manual/nat.rst]: please add a warning block: "NAT is not a security feature or firewall replacement!".

Maybe also mention that disabling PF will stop NAT as well.

File: source/manual/nat.rst (hunk 2 above)
Comment (MichaelDeciso) [reply source/manual/nat.rst]: I added the terms. What would you like me to say about transparent proxies?

Comment (fabianfrz) [reply source/manual/nat.rst]: Please simplify:

All transparent proxies in OPNsense are implemented in a way, that you need to DNAT a connection to a daemon listening on localhost, which tries to get the original desination IP from the `/dev/pf`device. The transparent proxy tries to connect to the original server depending on the proxy configuration.

For example if you want to to use a transparent proxy for HTTP, you need to forward traffic to from TCP/80 via IPv4  to 127.0.0.1:3128 in the default configuration.

Comment (MichaelDeciso) [reply source/manual/nat.rst]: Ok, I have a question: why does this fall under "port forwarding"/inbound NAT and not under outbound/SNAT? Is it not possible to act as a transparent proxy for LAN clients?

**Comment by MichaelDeciso** [comment]
This PR should be ready to go.

**Pull request merged**